src/test/groovy/org/web3j/solidity/gradle/plugin/SolidityCompilerTest.groovy

```diff
@@ -34,4 +38,18 @@ class SolidityCompilerTest {
         final def solc = new SolidityCompiler().getExecutable()
         assertThat(solc.name).isEqualTo("solc")
     }
-}
+
+    @Test(expected = RuntimeException.class)
+    void testWindowsDll()
+    {
+        System.setProperty("os.name","Windows 10")
+        final def solc = new SolidityCompiler();
+    }
+
+    @Test
+    void testFakeWindowsDll() {
+        final requiredFile = folder.newFile("vcruntime140.dll")
+        println requiredFile.getParent()
+        assertThat(SolidityCompiler.checkWindowsVcppExists("/var/;/etc/;" + requiredFile.getParent())).isEqualTo(true)
+    }
```

Comment:
This could be tested implicitly to keep the method private...

Reply:
I don't think there is an easy way around this the map returned by getenv is read-only.